### PR TITLE
bugfix(rest-api): Should return a 404 if MIME type not found

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

Missed this in the initial implementation. So not really a bugfix, but kinda. Anyway, simply wraps the request to get the provided MIME type in an `Option` and then goes from there. `Some(entry)` => Do as it did before (copy and paste); `None` => Return a response of `HttpStatus.SC_NOT_FOUND`.

#1306

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java [hunk 1]
```diff
@@ -38,6 +38,14 @@ public void testRetrieveMimeTypeConfiguration() throws Exception {
     }
   }
 
+  @Test
+  public void testMimeTypeConfigurationNotFoundHandling() throws IOException {
+    final String noSuchMimeType = "blah/blah";
+    final HttpMethod method =
+        new GetMethod(MIMETYPE_API_ENDPOINT + "/viewerconfig/" + noSuchMimeType);
```
Comment (SammyIsConfused) [inline autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:45]: "/viewerconfig/" is repeated in this file. Should be a constant.

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java [hunk 2]
```diff
@@ -38,6 +38,14 @@ public void testRetrieveMimeTypeConfiguration() throws Exception {
     }
   }
 
+  @Test
+  public void testMimeTypeConfigurationNotFoundHandling() throws IOException {
```
Comment (SammyIsConfused) [inline autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:42]: I'm assuming that it also handles the same for an empty string MIME type? Have you tested that?

**Review by SammyIsConfused — APPROVED** [review]
Looks good! Just a query about empty MIME types and a constant.

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java (hunk 1 above)
Comment (edalex-ian) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:45]: Ah yes! Thank you. There was one, now there are two. I'll rework.

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java (hunk 2 above)
Comment (edalex-ian) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:42]: I'll have a play. I am/was hoping the framework looks after that - as swagger shows them as mandatory, so I'm thinking it should know what's going on. But... not confident in that. :wink:

**Review by mrblippy — APPROVED** [review]

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java (hunk 2 above)
Comment (edalex-ian) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:42]: I have now tested, and yes. Seeing it's path vars, if you start dropping some etc then it triggers the 404. So all good.

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java (hunk 1 above)
Comment (edalex-ian) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java]: Doneski.

**Comment by edalex-ian** [comment]
Cool, just lookin' for green, and then I'll merge. :wink:

**Review by SammyIsConfused — APPROVED** [review]

**Pull request merged**